diff --git a/src/apply/apply.go b/src/apply/apply.go
index 47f1346..47badab 100644
--- a/src/apply/apply.go
+++ b/src/apply/apply.go
@@ -2,6 +2,7 @@ package apply
 
 import (
 	"fmt"
+	"errors"
 	"os"
 	"path/filepath"
 	"strings"
@@ -65,10 +66,17 @@ func AdditionalOptions(appsFolderPath string, flags Flag) {
 // To not use custom css, set `themeFolder` to blank string
 // To use default color scheme, set `scheme` to `nil`
 func UserCSS(appsFolderPath, themeFolder string, scheme map[string]string) {
+	colorsSrc := os.Getenv("COLORS_CSS_PATH")
 	colorsDest := filepath.Join(appsFolderPath, "xpui", "colors.css")
-	if err := os.WriteFile(colorsDest, []byte(getColorCSS(scheme)), 0700); err != nil {
+
+	if len(colorsSrc) == 0 {
+		utils.Fatal(errors.New("COLORS_CSS_PATH environment variable is not set"))
+	}
+
+	if err := os.Symlink(colorsSrc, colorsDest); err != nil {
 		utils.Fatal(err)
 	}
+
 	cssDest := filepath.Join(appsFolderPath, "xpui", "user.css")
 	if err := os.WriteFile(cssDest, []byte(getUserCSS(themeFolder)), 0700); err != nil {
 		utils.Fatal(err)
